sql/2019/07_Performance/07_17.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL
+# 07_17: Percentiles of blocking CSS requests
+#This metric comes from Lighthouse only
+CREATE TEMPORARY FUNCTION renderBlockingCSS(payload STRING)
+RETURNS NUMERIC LANGUAGE js AS '''
+  var obj = JSON.parse(payload); 

```

Comment:
Could make use of array methods to simplify this function. Also rely on the try/catch for error handling.


```sql
#standardSQL
# 07_17: Percentiles of blocking CSS requests
#This metric comes from Lighthouse only
CREATE TEMPORARY FUNCTION renderBlockingCSS(report STRING)
RETURNS NUMERIC LANGUAGE js AS '''
try {
  var $ = JSON.parse(report);
  return $.audits['render-blocking-resources'].details.items.filter(item => {
    return item.url.toLowerCase().includes('.css');
  }).length;
} catch (e) {
  return null;
}
''';

SELECT
  APPROX_QUANTILES(renderBlockingCSS, 1000)[OFFSET(100)] AS p10,
  APPROX_QUANTILES(renderBlockingCSS, 1000)[OFFSET(250)] AS p25,
  APPROX_QUANTILES(renderBlockingCSS, 1000)[OFFSET(500)] AS p50,
  APPROX_QUANTILES(renderBlockingCSS, 1000)[OFFSET(750)] AS p75,
  APPROX_QUANTILES(renderBlockingCSS, 1000)[OFFSET(900)] AS p90
FROM (
  SELECT renderBlockingCSS(report) AS renderBlockingCSS
  FROM `httparchive.lighthouse.2019_07_01_mobile`)
```

Reply:
Modified as suggested to use array methods to simplify the UDF, added try/catch. Learnt a better way of filtering.